### PR TITLE
Additions to the bigraded Betti number methods

### PR DESCRIPTION
**Pull request by OP5642**
This pull request expands the functionality of ``bigraded_betti_number()`` methods of a ``SimplicialComplex`` by adding an additional parameter ``verbose``. The ``verbose`` option increases verbosity while computing the invariant by printing the subcomplexes along with their non-trivial homology groups.

This PR is a part of #35640 (GSoC 2023).
 
### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by OP5642** [comment]
@tscrim

**Comment by tscrim** [comment]
This should not depend on #35875. Adding the bigraded methods should be done on that PR alone. These PRs are independent AFAICS.

I agree with the `verbose` option, but I don't agree with the `tablular`, at least using `print()`. If anything, it should return an actual table(-like) object. Although I am not convinced that it isn't something the user should just do on their own.

File: src/sage/topology/simplicial_complex.py [hunk 1]
```diff
@@ -4885,18 +4923,29 @@ def bigraded_betti_numbers(self, base_ring=ZZ):
...
-        of `(b-a-1)`-th (co)homologies of full subcomplexes with exactly `b` vertices.
+        Bigraded Betti number with indices `(-a, 2b)` is defined as a
+        sum of ranks of `(b-a-1)`-th (co)homologies of full subcomplexes
+        with exactly `b` vertices.
+
+        INPUT:
+
+        - ``base_ring`` -- (optional, default ``ZZ``) the base ring used
+          when computing homology
+        - ``verbose`` -- (optional, default ``False``) if ``True``, print
```
Comment (tscrim) [inline src/sage/topology/simplicial_complex.py]: You should add a note that this avoids the cache.

File: src/sage/topology/simplicial_complex.py [hunk 2]
```diff
@@ -4846,11 +4846,19 @@ def intersection(self, other):
...
+    def bigraded_betti_numbers(self, base_ring=ZZ, verbose=False):
         r"""
         Return a dictionary of the bigraded Betti numbers of ``self``,
         with keys `(-a, 2b)`.
 
+        INPUT:
+
+        - ``base_ring`` -- (optional, default ``ZZ``) the base ring used
+          when computing homology
+        - ``verbose`` -- (optional, default ``False``) if ``True``, print
```
Comment (tscrim) [inline src/sage/topology/simplicial_complex.py]: You should add a note that this avoids the cache. Also `(optional, default: X)` -> `(default: X)`.

File: src/sage/topology/simplicial_complex.py [hunk 3]
```diff
@@ -4863,8 +4871,38 @@ def bigraded_betti_numbers(self, base_ring=ZZ):
             [((0, 0), 1), ((-1, 6), 1), ((-1, 4), 2), ((-2, 8), 1), ((-2, 6), 1)]
             sage: sorted(Y.bigraded_betti_numbers(base_ring=QQ).items(), reverse=True)
             [((0, 0), 1), ((-1, 4), 3), ((-2, 8), 2), ((-2, 6), 1), ((-3, 10), 1)]
-        """
-        if base_ring in self._bbn_all_computed:
+
+        If we wish to view them in a form of a table, it is
+        simple enough to create a function as such::
```
Comment (tscrim) [inline src/sage/topology/simplicial_complex.py:4893]:
```suggestion
        We print the bigraded Betti numbers in a table::
```

**Review by tscrim — CHANGES_REQUESTED** [review]

File: src/sage/topology/simplicial_complex.py (hunk 2 above)
Comment (OP5642) [reply src/sage/topology/simplicial_complex.py]: Thank you.

**Comment by OP5642** [comment]
Thank you for the review, @tscrim. I also modified the ``verbose`` option for ``bigraded_betti_numbers()`` method by adding the index (in the form ``(-i, 2j``)) as a prefix, as well as one more doctest which demonstrates the verbose output. How should I deal with lines 4945 and 5024, which seem way too long?

**Review by tscrim — APPROVED** [review]
LGTM.

**Comment by vbraun** [comment]
Merge conflict

**Comment by tscrim** [comment]
@OP5642 Can you `git rebase` this over the most recent `develop` (and then force push it)?

**Pull request merged**